src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -80,7 +80,7 @@ sealed class HashMap[A, +B] extends AbstractMap[A, B]
 
   protected def elemHashCode(key: A) = key.##
 
-  protected final def improve(hcode: Int) = {
+  protected final def improve(hcode: Int) = { // TODO unify with HashUtils.improve
```

Comment:
I don't think we should change the hashing scheme in point releases for 2.12.  Leave it alone until 2.13.